nextcord/polls.py

```diff
@@ -0,0 +1,425 @@
+        )
+
+    def voters(
+        self, *, limit: Optional[int] = None, after: Optional[Snowflake] = None
+    ) -> AnswerVotersIterator:
+        """Returns an :class:`AsyncIterator` representing the users who have voted for this option.
+
+        The ``after`` parameter must represent an user
+        and meet the :class:`abc.Snowflake` abc.
+
+        Examples
+        --------
+        Usage ::
+
+            # I do not actually recommend doing this.
+
+            async for voter in count.voters():
+                await channel.send(f"{voter} has voted with choice number {count.id}!")
+
+        Flattening into a list ::
+
+            voters = await count.voters().flatten()
+            # voters is now a list of User...
+            await channel.send("Those people have voted with choice number {count.id}: {', '.join(voters)}")
+
+
+        Parameters
+        ----------
+        limit: Optional[:class:`int`]
+            The maximum number of results to return.
+            If not provided, returns all the users who
+            voted to the choice.
+        after: Optional[:class:`abc.Snowflake`]
+            For pagination, votes are sorted by member.
+        """
+        if limit is None:
+            limit = self.count
+
+        return AnswerVotersIterator(
+            message=self.poll.message, answer_id=self.id, after=after, limit=limit
+        )
+
+    def to_dict(self) -> PollAnswerCountPayload:
+        payload: PollAnswerCountPayload = {
+            "id": self.id,
+            "me_voted": self.me_voted,
+            "count": self.count,
+        }
+        return payload
+
+
+class PollResults:
+    """The result of the poll.
+
+    This contains the numbers of vote for each answer.
+
+    .. versionadded:: 3.0
+
+    Attributes
+    ----------
+    is_finalized: :class:`bool`
+        Whether the votes has been precisely counted.
+    answer_counts: List[:class:`PollAnswerCount`]
+        The counts for each answer.
+    poll: :class:`Poll`
+        The poll this poll result represents.
+    """
+
+    __slots__ = (
+        "is_finalized",
+        "answer_counts",
+        "poll",
+    )
+
+    def __init__(self, data: PollResultsPayload, poll: Poll) -> None:
+        self.is_finalized: bool = data["is_finalized"]
+        self.answer_counts: List[PollAnswerCount] = [
+            PollAnswerCount(answer_count, poll) for answer_count in data["answer_counts"]
+        ]
+        self.poll: Poll = poll
+
+    def __repr__(self) -> str:
+        return (
+            "<PollResults "
+            f"is_finalized={self.is_finalized!r} "
+            f"answer_counts={self.answer_counts!r} "
+            ">"
+        )
+
+    def to_dict(self) -> PollResultsPayload:
+        payload: PollResultsPayload = {
+            "is_finalized": self.is_finalized,
+            "answer_counts": [answer_count.to_dict() for answer_count in self.answer_counts],
+        }
+        return payload
+
+
+class PollCreateRequest:
```

Comment:
`PollCreateRequest` and `Poll` being seperate objects is kinda annoying, but I get it. If there is a clean way to merge them that would also be somewhat future proof that would be great, if not then ignore this.